File: frontend/src/app/dtos/locationResponse.ts
import { LocationResourceResponse } from "./locationResourceResponse";

export class LocationResponse {
    constructor(
        public uuid: string = '',
        public name: string | null = null,
        public description: string | null = null,
        public isPublic: boolean = true,
        public address: string | null = null,
        public screens: LocationResourceResponse[] = [],
        public clients: LocationResourceResponse[] = []
    ) {}
}

File: backend/src/main/java/com/jpromi/darts/backend/models/LocationResponse.java
package com.jpromi.darts.backend.models;

import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
@Builder
public class LocationResponse {
    private UUID uuid;
    private String name;
    private String description;
    private String address;
    private Boolean isPublic;
    private List<Screen> screens;
    private List<Client> clients;

    @Data
    @Builder
    public static class Client {
        private UUID uuid;
        private String name;
        private String token;
    }

    @Data
    @Builder
    public static class Screen {
        private UUID uuid;
        private String name;
        private String token;
    }
}
